### PR TITLE
fix bug where FourierCurrentPotentialField would not initialize Phi_mn coefficients properly

### PR DESCRIPTION
**Pull request by dpanici**
- also adds asserts to ensure passed-in modes match the coefficients size to `FourierCurrentPotentialField` and `FourierRZToroidalSurface`

### CODE REVIEW

**Comment by dpanici** [comment]
> Was the bug was caused by a size mismatch between `Phi_mn` and `modes_Phi`? It might be good to include an explicit check for that, with a warning/error for the user. We should also probably do that check for other classes that are initialized with both coefficient values and mode numbers, like `FourierRZToroidalSurface`. Otherwise this looks good.

It was not with a size mismatch from the provided modes, it was that the initial modes provided did not match the full modes array of the corresponding Basis created (i.e. the copy coeffs line was missing). I used code from that surface class you mention to fix the issue. Ensuring they are the same size initially is a good idea though too, I will add

File: desc/geometry/curve.py [hunk 1]
```diff
@@ -533,6 +540,8 @@ def __init__(
         else:
             modes = np.asarray(modes)
         assert issubclass(modes.dtype.type, np.integer)
+        print(modes.shape)
```
Comment (f0uriest) [inline desc/geometry/curve.py:543]: extra print

File: desc/geometry/curve.py [hunk 2]
```diff
@@ -72,6 +72,9 @@ def __init__(
 
         modes_R, modes_Z = np.asarray(modes_R), np.asarray(modes_Z)
 
+        assert R_n.size == modes_R.size, "R_n size and modes_R must be the same size!"
```
Comment (f0uriest) [inline desc/geometry/curve.py:75]: I don't think we need exclamation points on all of these

Comment (ddudt) [reply desc/geometry/curve.py:75]: we should write all of our error messages with a really angry tone so the users feel stupid when they do something wrong

**Review by ddudt — APPROVED** [review]
Looks good, besides Rory's comments

**Review by f0uriest — APPROVED** [review]

**Review by ddudt — APPROVED** [review]

**Pull request merged**